crates/chain-config/src/config/state.rs

```diff
@@ -59,3 +189,254 @@ pub trait ChainConfigDb {
+    }
+
+    #[test]
+    fn snapshot_contract_with_balances() {
+        let config = test_config_contract(false, true, false, false);
+        let json = serde_json::to_string_pretty(&config).unwrap();
+        insta::assert_snapshot!(json);
+    }
+
+    #[test]
+    fn can_roundtrip_contract_with_balances() {
+        let config = test_config_contract(false, true, false, false);
+        let json = serde_json::to_string(&config).unwrap();
+        let deserialized_config: StateConfig =
+            serde_json::from_str(json.as_str()).unwrap();
+        assert_eq!(config, deserialized_config);
+    }
+
+    #[test]
+    fn snapshot_contract_with_utxo_id() {
+        let config = test_config_contract(false, false, true, false);
+        let json = serde_json::to_string_pretty(&config).unwrap();
+        insta::assert_snapshot!(json);
+    }
+
+    #[test]
+    fn can_roundtrip_contract_with_utxoid() {
+        let config = test_config_contract(false, false, true, false);
+        let json = serde_json::to_string(&config).unwrap();
+        let deserialized_config: StateConfig =
+            serde_json::from_str(json.as_str()).unwrap();
+        assert_eq!(config, deserialized_config);
+    }
+
+    #[test]
+    fn snapshot_contract_with_tx_pointer() {
+        let config = test_config_contract(false, false, false, true);
+        let json = serde_json::to_string_pretty(&config).unwrap();
+        insta::assert_snapshot!(json);
+    }
+
+    #[test]
+    fn can_roundtrip_contract_with_tx_pointer() {
+        let config = test_config_contract(false, false, false, true);
+        let json = serde_json::to_string(&config).unwrap();
+        let deserialized_config: StateConfig =
+            serde_json::from_str(json.as_str()).unwrap();
+        assert_eq!(config, deserialized_config);
+    }
+
+    #[test]
+    fn snapshot_simple_coin_state() {
+        let config = test_config_coin_state();
+        let json = serde_json::to_string_pretty(&config).unwrap();
+        insta::assert_snapshot!(json);
+    }
+
+    #[test]
+    fn can_roundtrip_simple_coin_state() {
+        let config = test_config_coin_state();
+        let json = serde_json::to_string(&config).unwrap();
+        let deserialized_config: StateConfig =
+            serde_json::from_str(json.as_str()).unwrap();
+        assert_eq!(config, deserialized_config);
+    }
+
+    #[test]
+    fn snapshot_simple_message_state() {
+        let config = test_message_config();
+        let json = serde_json::to_string_pretty(&config).unwrap();
+        insta::assert_snapshot!(json);
+    }
+
+    #[test]
+    fn can_roundtrip_simple_message_state() {
+        let config = test_message_config();
+        let json = serde_json::to_string(&config).unwrap();
+        let deserialized_config: StateConfig =
+            serde_json::from_str(json.as_str()).unwrap();
+        assert_eq!(config, deserialized_config);
+    }
+
+    fn test_config_contract(
```

Comment:
It doesn't look like any of these fields are used at the same time. It might make more sense to move the shared code out to a `base_state_config()` and create a small helper function for each case. e.g. 
```
fn state_config_with_state() -> StateConfig {
    let test_key: Bytes32 = rng.gen();
    let test_value: Bytes32 = rng.gen();
    let state = Some(vec![(test_key, test_value)]);
    StateConfig {
        state,
        ..base_state_config()
    }
}
```
What you have would also be more readable as a builder pattern, but I think that's more complexity than is needed here.